diff --git a/node_modules/@sentry/react-native/ios/RNSentryReplay.mm b/node_modules/@sentry/react-native/ios/RNSentryReplay.mm
index 40575a9..e051770 100644
--- a/node_modules/@sentry/react-native/ios/RNSentryReplay.mm
+++ b/node_modules/@sentry/react-native/ios/RNSentryReplay.mm
@@ -2,7 +2,6 @@
 #import "RNSentryReplayBreadcrumbConverterHelper.h"
 #import "RNSentryReplayQuality.h"
 #import "RNSentryVersion.h"
-#import "React/RCTTextView.h"
 #import "Replay/RNSentryReplayMask.h"
 #import "Replay/RNSentryReplayUnmask.h"
 #import <Sentry/PrivateSentrySDKOnly.h>
