diff --git a/node_modules/react-native-nitro-sqlite/cpp/operations.cpp b/node_modules/react-native-nitro-sqlite/cpp/operations.cpp
index 0655754..88e80cd 100644
--- a/node_modules/react-native-nitro-sqlite/cpp/operations.cpp
+++ b/node_modules/react-native-nitro-sqlite/cpp/operations.cpp
@@ -165,7 +165,7 @@ SQLiteExecuteQueryResult sqliteExecute(const std::string& dbName, const std::str
           switch (column_type) {
 
             case SQLITE_INTEGER: {
-              auto column_value = sqlite3_column_int64(statement, i);
+              auto column_value = sqlite3_column_double(statement, i);
               row[column_name] = column_value;
               break;
             }