diff --git a/node_modules/react-native-worklets/RNWorklets.podspec b/node_modules/react-native-worklets/RNWorklets.podspec
index d658b49..6e8851a 100644
--- a/node_modules/react-native-worklets/RNWorklets.podspec
+++ b/node_modules/react-native-worklets/RNWorklets.podspec
@@ -71,6 +71,7 @@ Pod::Spec.new do |s|
       '"$(PODS_ROOT)/DoubleConversion"',
       '"$(PODS_ROOT)/Headers/Private/React-Core"',
       '"$(PODS_ROOT)/Headers/Private/Yoga"',
+      '"$(PODS_ROOT)/../../node_modules/react-native/ReactCommon/jsitooling"',
     ].join(' '),
     "FRAMEWORK_SEARCH_PATHS" => '"${PODS_CONFIGURATION_BUILD_DIR}/React-hermes"',
     "CLANG_CXX_LANGUAGE_STANDARD" => "c++20",
