diff --git a/node_modules/react-native-change-icon/android/build.gradle b/node_modules/react-native-change-icon/android/build.gradle
index aa5226c..bf54025 100644
--- a/node_modules/react-native-change-icon/android/build.gradle
+++ b/node_modules/react-native-change-icon/android/build.gradle
@@ -4,7 +4,7 @@ buildscript {

   repositories {
     google()
-    jcenter()
+    mavenCentral()
   }

   dependencies {
@@ -52,7 +52,6 @@ android {

 repositories {
   mavenCentral()
-  jcenter()
   google()

   def found = false
diff --git a/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt b/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt
index 70c3e2c..724b3fe 100644
--- a/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt
+++ b/node_modules/react-native-change-icon/android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt
@@ -21,7 +21,7 @@ class ChangeIconModule(reactContext: ReactApplicationContext, private val packag

     @ReactMethod
     fun changeIcon(enableIcon: String, promise: Promise) {
-        val activity: Activity? = currentActivity
+        val activity: Activity? = reactApplicationContext.currentActivity
         if (activity == null || enableIcon.isEmpty()) {
             promise.reject("Icon string is empty.")
             return
@@ -46,7 +46,7 @@ class ChangeIconModule(reactContext: ReactApplicationContext, private val packag

     private fun completeIconChange() {
         if (iconChanged) {
-          val activity: Activity? = currentActivity
+          val activity: Activity? = reactApplicationContext.currentActivity
           if (activity == null) {
             return
           }
