diff --git a/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug
new file mode 100644
index 0000000..a874a0d
--- /dev/null
+++ b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug
@@ -0,0 +1,18 @@
+{
+  "installationFolder": "/Users/michal/Source/minibits_wallet/node_modules/react-native-worklets/android/build/intermediates/prefab_package/debug/prefab",
+  "gradlePath": ":react-native-worklets",
+  "packageInfo": {
+    "packageName": "react-native-worklets",
+    "packageVersion": "0.6.0",
+    "packageSchemaVersion": 2,
+    "packageDependencies": [],
+    "modules": [
+      {
+        "moduleName": "worklets",
+        "moduleHeaders": "/Users/michal/Source/minibits_wallet/node_modules/react-native-worklets/android/build/prefab-headers/worklets",
+        "moduleExportLibraries": [],
+        "abis": []
+      }
+    ]
+  }
+}
\ No newline at end of file
diff --git a/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release
new file mode 100644
index 0000000..c93fcb5
--- /dev/null
+++ b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release
@@ -0,0 +1,18 @@
+{
+  "installationFolder": "/Users/michal/Source/minibits_wallet/node_modules/react-native-worklets/android/build/intermediates/prefab_package/release/prefab",
+  "gradlePath": ":react-native-worklets",
+  "packageInfo": {
+    "packageName": "react-native-worklets",
+    "packageVersion": "0.6.0",
+    "packageSchemaVersion": 2,
+    "packageDependencies": [],
+    "modules": [
+      {
+        "moduleName": "worklets",
+        "moduleHeaders": "/Users/michal/Source/minibits_wallet/node_modules/react-native-worklets/android/build/prefab-headers/worklets",
+        "moduleExportLibraries": [],
+        "abis": []
+      }
+    ]
+  }
+}
\ No newline at end of file
diff --git a/node_modules/react-native-worklets/compatibility.json b/node_modules/react-native-worklets/compatibility.json
index 850b8c4..a777247 100644
--- a/node_modules/react-native-worklets/compatibility.json
+++ b/node_modules/react-native-worklets/compatibility.json
@@ -3,7 +3,7 @@
     "react-native": ["0.78", "0.79", "0.80", "0.81"]
   },
   "0.6.x": {
-    "react-native": ["0.78", "0.79", "0.80", "0.81"]
+    "react-native": ["0.78", "0.79", "0.80", "0.81", "0.82"]
   },
   "0.5.x": {
     "react-native": ["0.78", "0.79", "0.80", "0.81"]
